lib/chef/resource/chocolatey_installer.rb

```diff
@@ -0,0 +1,198 @@
+
+      property :chocolatey_version, String,
+        description: "Specifies a target version of Chocolatey to install. By default, the latest stable version is installed. This will use the value in $env:ChocolateyVersion by default, if that environment variable is present. This parameter is ignored if download_url is set."
+
+      property :use_native_unzip, [TrueClass, FalseClass], default: false,
+        description: "If set, uses built-in Windows decompression tools instead of 7zip when unpacking the downloaded nupkg. This will be set by default if use_native_unzip is set to a value other than 'false' or '0'. This parameter will be ignored in PS 5+ in favour of using the Expand-Archive built in PowerShell cmdlet directly."
+
+      property :ignore_proxy, [TrueClass, FalseClass], default: false,
+        description: "If set, ignores any configured proxy. This will override any proxy environment variables or parameters. This will be set by default if ignore_proxy is set to a value other than 'false' or '0'."
+
+      property :proxy_url, String,
+        description: "Specifies the proxy URL to use during the download."
+
+      property :proxy_user, String,
+        description: "The username to use to build a proxy credential with. Will be consumed by the proxy_credential property if both this property and proxy_password are set"
+
+      property :proxy_password, String,
+        description: "The password to use to build a proxy credential with. Will be consumed by the proxy_credential property if both this property and proxy_user are set"
+
+      load_current_value do
+        current_state = is_choco_installed?
+        current_value_does_not_exist! if current_state == false
+        current_state
+      end
+
+      def is_choco_installed?
+        ::File.exist?("#{ENV["ALLUSERSPROFILE"]}\\chocolatey\\bin\\choco.exe")
+      end
+
+      def get_choco_version
+        powershell_exec("choco --version").result
+      end
+
+      def existing_version
+        Gem::Version.new(get_choco_version)
+      end
+
+      def define_resource_requirements
+        [ new_resource.proxy_user, new_resource.proxy_password ].each do
+          requirements.assert(:install, :upgrade).each do |a|
+            a.assertion do
+              (!new_resource.proxy_user.nil? && new_resource.proxy_password.nil?) || (new_resource.proxy_user.nil? && !new_resource.proxy_password.nil?)
+            end
+            a.failure_message(Chef::Exceptions::ValidationFailed, "You must specify both a proxy_user and a proxy_password")
+            a.whyrun("Assuming that if you have configured a 'proxy_user' you must also supply a 'proxy_password'")
+          end
+        end
+      end
+
+      action :install, description: "Installs Chocolatey package manager" do
+        unless new_resource.download_url.nil?
```

Comment:
You can just do
```suggestion
        if new_resource.download_url
```

Ruby treats that was defined, not nil, not false.

Reply:
done